File: platforms/agl/alexa-voiceagent-service/src/plugins/aac-services/aasb/aac-platform/alexa/src/AlertsHandler.cpp
/*
 * Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
 *
 * Licensed under the Apache License, Version 2.0 (the "License").
 * You may not use this file except in compliance with the License.
 * A copy of the License is located at
 *
 *     http://aws.amazon.com/apache2.0/
 *
 * or in the "license" file accompanying this file. This file is distributed
 * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
 * express or implied. See the License for the specific language governing
 * permissions and limitations under the License.
 */

#include "AlertsHandler.h"

#include <sstream>
#include <aasb/Consts.h>
#include "PlatformSpecificLoggingMacros.h"

/**
 * Specifies the severity level of a log message
 * @sa @c aace::logger::LoggerEngineInterface::Level
 */
using Level = aace::logger::LoggerEngineInterface::Level;

namespace aasb {
namespace alexa {

const std::string TAG = "aasb::alexa::AlertsHandler";

std::shared_ptr<AlertsHandler> AlertsHandler::create(
    std::shared_ptr<aasb::core::logger::LoggerHandler> logger,
    std::weak_ptr<aasb::bridge::ResponseDispatcher> responseDispatcher) {
    auto alertsHandler = std::shared_ptr<AlertsHandler>(new AlertsHandler(logger, responseDispatcher));
    return alertsHandler;
}

AlertsHandler::AlertsHandler(
    std::shared_ptr<aasb::core::logger::LoggerHandler> logger,
    std::weak_ptr<aasb::bridge::ResponseDispatcher> responseDispatcher) :
        m_logger(logger), m_responseDispatcher(responseDispatcher) {
}

void AlertsHandler::alertStateChanged(const std::string &alertToken, Alerts::AlertState state, const std::string &reason) {
    std::stringstream ss;
    ss << state;
    m_logger->log(Level::INFO, TAG, "alertChanged: " + alertToken + " state: " + ss.str() + " reason: " + reason);
}

void AlertsHandler::alertCreated(const std::string &alertToken, const std::string &detailedInfo) {
    m_logger->log(Level::INFO, TAG, "alertCreated: " + alertToken + " info: " + detailedInfo);
}

void AlertsHandler::alertDeleted(const std::string &alertToken) {
    m_logger->log(Level::INFO, TAG, "alertDeleted: " + alertToken);
}

} // namespace alexa
} // namespace aasb

File: samples/android/app/src/main/java/com/amazon/sampleapp/impl/Alerts/AlertsHandler.java
/*
 * Copyright 2017-2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
 *
 * Licensed under the Apache License, Version 2.0 (the "License").
 * You may not use this file except in compliance with the License.
 * A copy of the License is located at
 *
 *     http://aws.amazon.com/apache2.0/
 *
 * or in the "license" file accompanying this file. This file is distributed
 * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
 * express or implied. See the License for the specific language governing
 * permissions and limitations under the License.
 */

package com.amazon.sampleapp.impl.Alerts;

import android.app.Activity;
import android.view.View;
import android.widget.TextView;

import com.amazon.aace.alexa.Alerts;
import com.amazon.sampleapp.R;
import com.amazon.sampleapp.impl.Logger.LoggerHandler;

public class AlertsHandler extends Alerts {
    private static final String sTag = "Alerts";

    private final Activity mActivity;
    private final LoggerHandler mLogger;
    private TextView mStateText;

    public AlertsHandler(Activity activity, LoggerHandler logger) {
        mActivity = activity;
        mLogger = logger;
        setupGUI();
    }

    @Override
    public void alertStateChanged(final String alertToken, final AlertState state, final String reason) {
        mLogger.postInfo(sTag,
                String.format("Alert State Changed. STATE: %s, REASON: %s, TOKEN: %s", state, reason, alertToken));
        mActivity.runOnUiThread(new Runnable() {
            @Override
            public void run() {
                mStateText.setText(state != null ? state.toString() : "");
            }
        });
    }

    @Override
    public void alertCreated(String alertToken, String detailedInfo) {
        mLogger.postInfo(
                sTag, String.format("Alert Created. TOKEN: %s, Detailed Info payload: %s", alertToken, detailedInfo));
    }

    @Override
    public void alertDeleted(String alertToken) {
        mLogger.postInfo(sTag, String.format("Alert Deleted. TOKEN: %s", alertToken));
    }

    private void onLocalStop() {
        mLogger.postInfo(sTag, "Stopping active alert");
        super.localStop();
    }

    private void onRemoveAllAlerts() {
        mLogger.postInfo(sTag, "Removing all pending alerts from storage");
        super.removeAllAlerts();
    }

    private void setupGUI() {
        mStateText = mActivity.findViewById(R.id.alertState);

        mActivity.findViewById(R.id.stopAlertButton).setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                onLocalStop();
            }
        });
    }
}
